src/esp/scene/SemanticScene.cpp

```diff
@@ -236,46 +245,104 @@ void SemanticScene::buildSemanticOBBsFromCCs(
 
     if ((semanticCCsPerID == perIDMapOfCCSemanticObjs.end()) ||
         (semanticCCsPerID->second.empty())) {
-      ESP_DEBUG() << msgPrefix << "\n!!!!!!Note : ID :" << objIdx
-                  << "has no corresponding CC; therefore, this semantic object "
-                     "will have no BBox since its color is not present in the "
-                     "vertices on the mesh.";
+      // keep a record of semantic IDs without any corresponding verts
+      unMappedObjectIDXs.emplace_back(objIdx);
+      // ESP_DEBUG() << "\n\t\t!!!!!!" << msgPrefix
+      //             << "Note : Semantic Scene Annotation ID :" << objIdx
+      //             << "is not present in the mesh - there are no vertices "
+      //             << "with this annotation's color assigned to them; "
+      //             << "therefore, this semantic object will have no BBox.";
       continue;
     }
 
     std::vector<CCSemanticObject::ptr> vecOfCCSemanticObjs =
         semanticCCsPerID->second;
-
     ESP_VERY_VERBOSE() << Cr::Utility::formatString(
-        "vec size {} Displaying elements in decreasing order "
-        "for objIDX : {} | name : {} | vol : {}",
-        vecOfCCSemanticObjs.size(), objIdx, ssdObj.id(), ssdObj.obb().volume());
+        "{}Semantic CC vec size {} Displaying elements in decreasing order "
+        "for objIDX : {} | name : {} | current (pre-CC calc) vol : {} | "
+        "Fraction of max vol to use "
+        "{}",
+        msgPrefix, vecOfCCSemanticObjs.size(), objIdx, ssdObj.id(),
+        ssdObj.obb().volume(), maxVolFraction);
     if (vecOfCCSemanticObjs.size() == 1) {
-      // if only one element, just use specified element
+      // if only one element, always use specified CC's bbox, built off all
+      // verts with semantic annotation
       ssdObj.setObb(vecOfCCSemanticObjs[0]->obb());
     } else {
+      // Multiple elements, use some fraction of CCs based on volume
       // build temp multimap keyed by volume
       perSemanticObjCCs.clear();
-      for (const auto& CCObj : vecOfCCSemanticObjs) {
-        perSemanticObjCCs.emplace(CCObj->obb().volume(), CCObj);
+      semanticObjVolToSetIDX.clear();
```

Comment:
Can't these also be reserved with the right size?

Reply:
Not in a multimap, afaik.  Ordered container, probably backed by a bst, so node-based.  Not sure how you would pre-allocate such a structure.

Reply:
@jturner65 Ah, I see, it's purposely an ordered multimap as opposed to an unordered one? Makes sense.